service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Job.java

```diff
@@ -22,59 +23,27 @@
     private String jobId;
     private JobClient jobClient;
 
-    /**
-     * CONSTRUCTOR
-     *
-     * @param connectionString is the IoTHub connection string.
-     * @throws IOException This exception is thrown if the object creation failed
-     */
-    Job(String connectionString) throws IOException
+    Job(String connectionString)
     {
         if ((connectionString == null) || connectionString.isEmpty())
         {
-            /* Codes_SRS_JOB_21_001: [The constructor shall throw IllegalArgumentException if the input string is null or empty.] */
             throw new IllegalArgumentException("Connection string cannot be null or empty");
         }
 
-        /* Codes_SRS_JOB_21_003: [If no jobId is provided, the constructor shall generate a unique jobId to identify the Job in the Iothub.] */
```

Comment:
This makes it sound like jobId should be a input parameter too. But looks like in all Job constructors, we don't take it as input. Do we need that as an option for customers to send?

Reply:
Customers don't create instances of ```Job```, so we never need to take it as an input. We only create a value for it from a GUID so that when we schedule a twin update or a method invocation, we have a way to track that job by Id in later calls to jobClient.getJob(...).